include/crypto/aes_engine.hpp

```diff
@@ -0,0 +1,167 @@
+#pragma once
+
+#include <array>
+#include <cstring>
+#include <cstdint>
+#include <climits>
+#include <filesystem>
+#include <optional>
+
+#include "helpers.hpp"
+
+namespace Crypto {
+	constexpr std::size_t AesKeySize = 0x10;
+
+	using AESKey = std::array<uint8_t, AesKeySize>;
+
+	template<std::size_t N>
+	static std::array<uint8_t, N> rolArray(const std::array<uint8_t, N>& value, std::size_t bits) {
```

Comment:
All of these `const std::array<uint8_t, N>&` can be `std::span<const u8, N>` instead so it can take any container of randomly-addressable continuous elements.

Reply:
The std::span method actually is likely less intuitive here actually, I think

```cc
constexpr void updateNormalKey(std::size_t slotId) {
			if (m_generator.has_value() && hasKeyX(slotId) && hasKeyY(slotId)) {
				auto &keySlot = m_slots.at(slotId);
				AESKey keyX = keySlot.keyX.value();
				AESKey keyY = keySlot.keyY.value();

				keySlot.normalKey = rolArray(addArray(xorArray(rolArray(keyX, 2), keyY), m_generator.value()), 87);
			}
		}
		```
		The last line has a sort of snowball to process the array which works a lot better with arrays vs spans imo
		I think we should keep it, at least for now.